---
title: Resources
nav: true
---

    

# **Data analysis workflow**
    
### [GWAS-eQTL colocalization analysis workflow](https://hanruizhang.github.io/GWAS-eQTL-Colocalization/) by **Fang Li**
### [RNA-seq data analysis workflow using _salmon_](https://hanruizhang.github.io/RNAseq-analysis-workflow/) by Hanrui
### [RNA-seq analysis in Galaxy and visualization using UCSC genome browser and IGV](/file/CSHL_TGB2021.html) by Hanrui
### [Pathway and network analysis workflow](https://hanruizhang.github.io/Pathway-Network-Analysis/) by Hanrui  
### Functional genomic tools and fundamentals ([HTML](/file/Functional-Genomics-Tools_202004.html)) by **Jianting**
### Single cell RNA-seq analysis using Seurat 2.3.4 ([Rmd](/file/scRNAseq_Tutorial_latest.Rmd))/([HTML](/file/scRNAseq_Tutorial_latest.html)), Seurat 3.0 ([Rmd](/file/Zhang_Seurat_PBMC_20191001.Rmd)), and Seurat 3.1 ([Rmd](/file/00_Seurat_PBMC_20200310.Rmd))/([HTML](/file/00_Seurat_PBMC_20200310.html)) by **Sophie** and Hanrui   
### [HPC Instructions](https://hanruizhang.github.io/ZhangLab-HPC-Instructions/) by Hanrui
      
------
&nbsp;
# **Protocols**
### [**Human induced pluripotent stem cell to macrophage differentiation**](https://currentprotocols.onlinelibrary.wiley.com/doi/full/10.1002/cpsc.74)

We published the **[protocol](https://currentprotocols.onlinelibrary.wiley.com/doi/full/10.1002/cpsc.74)** to differentiate human induced pluripotent stem cell (iPSC) to macrophages.  iPSC-derived macrophages (IPSDM) obtained using this protocol have demonstrated excellent functional and transcriptomic fidelity relative to human monocyte-derived macrophages (HMDM), with advantages and successful applications in disease modeling using patients-derived and CRISPR-edited iPSC lines ([ATVB 2017](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5659288)). Through deep RNA-sequencing, the coding transcriptome ([Circ Res 2015](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4565503/)), alternative splicing events ([ATVB 2016](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4919157/)) and long non-coding RNA profiles ([JAHA 2017](https://www.ahajournals.org/doi/full/10.1161/JAHA.117.007431?url_ver=Z39.88-2003&rfr_id=ori:rid:crossref.org&rfr_dat=cr_pub%3dpubmed)) between isogenic IPSDM and HMDM at baseline and during activation are characterized, providing a comprehensive resource for planning IPSDM studies to model such events. This protocol and the associated resources provide a unique platform to understand human macrophage-specific functional, transcriptomic and genomic features in physiology and a broad spectrum of macrophage-relevant diseases.

The RNA-seq data of isogenic IPSDM and HMDM can be visualized in the UCSC genome browser by adding track hub using URL **https://de.cyverse.org/anon-files/iplant/home/chenyix/trackhub/GENEiPS_scaled/hub.txt** or through the following **[link](http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&hubUrl=https://de.cyverse.org/anon-files/iplant/home/chenyix/trackhub/GENEiPS_scaled/hub.txt)**. The original RNA-seq data are available from the NCBI Gene Expression Omnibus (GEO) under the accession number [GSE55536](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE55536).

**Notes:** Some users may find access to the above URL site is intermittent. Please try at a different time or contact hz2418@cumc.columbia.edu if you continue to have problem.

------
&nbsp;
# **Other Resources**
#### [The PISCES pipeline for protein activity inference in single cells](https://github.com/califano-lab/single-cell-pipeline/wiki/The-pipeline-for-Protein-Activity-Inference-in-Single-Cells-(PISCES)).   
#### [Data Science Boot Camps by DSI@Columbia](https://github.com/DS-BootCamp-DSI-Columbia)
#### [Applied Machine Learning](https://www.youtube.com/channel/UCMEXgDffQy6nS2a74Gby8ZA/videos) open lectures by Andreas Mueller at Columbia Data Science Institute
#### [HarvardX Biomedical Data Science Open Online Training](https://rafalab.github.io/pages/harvardx.html)
#### [Software Capentry Lessons](https://software-carpentry.org/lessons/)
#### [Data Carpentry Lessons](https://datacarpentry.org/lessons/#genomics-workshop)
#### [R for Data Science](https://r4ds.had.co.nz/index.html) by Garrett Grolemund and Hadley Wickham
#### [Python Data Science Handbook](https://jakevdp.github.io/PythonDataScienceHandbook/) by Jake VanderPlas
#### [Teaching materials at the Harvard Chan Bioinformatics Core](https://github.com/hbctraining)
#### [How to make a html webpage](https://columbia.hosted.panopto.com/Panopto/Pages/Viewer.aspx?id=18de7900-690b-4d1f-9344-acec015b6436) by Columbia Foundations for Research Computing.
#### And finally videos for [full body pilates](https://www.youtube.com/watch?v=sKIqLUlPGBw) and [30 days of Yoga](https://www.youtube.com/watch?v=oBu-pQG6sTY&list=RDQMkJ0HOrXEwlk&start_radio=1)

  


