<!DOCTYPE html>
<html>

<head>

<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>CSHL_TGB2021</title>


<style type="text/css">
body {
  font-family: Helvetica, arial, sans-serif;
  font-size: 14px;
  line-height: 1.6;
  padding-top: 10px;
  padding-bottom: 10px;
  background-color: white;
  padding: 30px; }

body > *:first-child {
  margin-top: 0 !important; }
body > *:last-child {
  margin-bottom: 0 !important; }

a {
  color: #4183C4; }
a.absent {
  color: #cc0000; }
a.anchor {
  display: block;
  padding-left: 30px;
  margin-left: -30px;
  cursor: pointer;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0; }

h1, h2, h3, h4, h5, h6 {
  margin: 20px 0 10px;
  padding: 0;
  font-weight: bold;
  -webkit-font-smoothing: antialiased;
  cursor: text;
  position: relative; }

h1:hover a.anchor, h2:hover a.anchor, h3:hover a.anchor, h4:hover a.anchor, h5:hover a.anchor, h6:hover a.anchor {
  background: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA09pVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMy1jMDExIDY2LjE0NTY2MSwgMjAxMi8wMi8wNi0xNDo1NjoyNyAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvIiB4bWxuczp4bXBNTT0iaHR0cDovL25zLmFkb2JlLmNvbS94YXAvMS4wL21tLyIgeG1sbnM6c3RSZWY9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9zVHlwZS9SZXNvdXJjZVJlZiMiIHhtcDpDcmVhdG9yVG9vbD0iQWRvYmUgUGhvdG9zaG9wIENTNiAoMTMuMCAyMDEyMDMwNS5tLjQxNSAyMDEyLzAzLzA1OjIxOjAwOjAwKSAgKE1hY2ludG9zaCkiIHhtcE1NOkluc3RhbmNlSUQ9InhtcC5paWQ6OUM2NjlDQjI4ODBGMTFFMTg1ODlEODNERDJBRjUwQTQiIHhtcE1NOkRvY3VtZW50SUQ9InhtcC5kaWQ6OUM2NjlDQjM4ODBGMTFFMTg1ODlEODNERDJBRjUwQTQiPiA8eG1wTU06RGVyaXZlZEZyb20gc3RSZWY6aW5zdGFuY2VJRD0ieG1wLmlpZDo5QzY2OUNCMDg4MEYxMUUxODU4OUQ4M0REMkFGNTBBNCIgc3RSZWY6ZG9jdW1lbnRJRD0ieG1wLmRpZDo5QzY2OUNCMTg4MEYxMUUxODU4OUQ4M0REMkFGNTBBNCIvPiA8L3JkZjpEZXNjcmlwdGlvbj4gPC9yZGY6UkRGPiA8L3g6eG1wbWV0YT4gPD94cGFja2V0IGVuZD0iciI/PsQhXeAAAABfSURBVHjaYvz//z8DJYCRUgMYQAbAMBQIAvEqkBQWXI6sHqwHiwG70TTBxGaiWwjCTGgOUgJiF1J8wMRAIUA34B4Q76HUBelAfJYSA0CuMIEaRP8wGIkGMA54bgQIMACAmkXJi0hKJQAAAABJRU5ErkJggg==) no-repeat 10px center;
  text-decoration: none; }

h1 tt, h1 code {
  font-size: inherit; }

h2 tt, h2 code {
  font-size: inherit; }

h3 tt, h3 code {
  font-size: inherit; }

h4 tt, h4 code {
  font-size: inherit; }

h5 tt, h5 code {
  font-size: inherit; }

h6 tt, h6 code {
  font-size: inherit; }

h1 {
  font-size: 28px;
  color: black; }

h2 {
  font-size: 24px;
  border-bottom: 1px solid #cccccc;
  color: black; }

h3 {
  font-size: 18px; }

h4 {
  font-size: 16px; }

h5 {
  font-size: 14px; }

h6 {
  color: #777777;
  font-size: 14px; }

p, blockquote, ul, ol, dl, li, table, pre {
  margin: 15px 0; }

hr {
  background: transparent url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAYAAAAECAYAAACtBE5DAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAAyJpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvIiB4bWxuczp4bXBNTT0iaHR0cDovL25zLmFkb2JlLmNvbS94YXAvMS4wL21tLyIgeG1sbnM6c3RSZWY9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9zVHlwZS9SZXNvdXJjZVJlZiMiIHhtcDpDcmVhdG9yVG9vbD0iQWRvYmUgUGhvdG9zaG9wIENTNSBNYWNpbnRvc2giIHhtcE1NOkluc3RhbmNlSUQ9InhtcC5paWQ6OENDRjNBN0E2NTZBMTFFMEI3QjRBODM4NzJDMjlGNDgiIHhtcE1NOkRvY3VtZW50SUQ9InhtcC5kaWQ6OENDRjNBN0I2NTZBMTFFMEI3QjRBODM4NzJDMjlGNDgiPiA8eG1wTU06RGVyaXZlZEZyb20gc3RSZWY6aW5zdGFuY2VJRD0ieG1wLmlpZDo4Q0NGM0E3ODY1NkExMUUwQjdCNEE4Mzg3MkMyOUY0OCIgc3RSZWY6ZG9jdW1lbnRJRD0ieG1wLmRpZDo4Q0NGM0E3OTY1NkExMUUwQjdCNEE4Mzg3MkMyOUY0OCIvPiA8L3JkZjpEZXNjcmlwdGlvbj4gPC9yZGY6UkRGPiA8L3g6eG1wbWV0YT4gPD94cGFja2V0IGVuZD0iciI/PqqezsUAAAAfSURBVHjaYmRABcYwBiM2QSA4y4hNEKYDQxAEAAIMAHNGAzhkPOlYAAAAAElFTkSuQmCC) repeat-x 0 0;
  border: 0 none;
  color: #cccccc;
  height: 4px;
  padding: 0;
}

body > h2:first-child {
  margin-top: 0;
  padding-top: 0; }
body > h1:first-child {
  margin-top: 0;
  padding-top: 0; }
  body > h1:first-child + h2 {
    margin-top: 0;
    padding-top: 0; }
body > h3:first-child, body > h4:first-child, body > h5:first-child, body > h6:first-child {
  margin-top: 0;
  padding-top: 0; }

a:first-child h1, a:first-child h2, a:first-child h3, a:first-child h4, a:first-child h5, a:first-child h6 {
  margin-top: 0;
  padding-top: 0; }

h1 p, h2 p, h3 p, h4 p, h5 p, h6 p {
  margin-top: 0; }

li p.first {
  display: inline-block; }
li {
  margin: 0; }
ul, ol {
  padding-left: 30px; }

ul :first-child, ol :first-child {
  margin-top: 0; }

dl {
  padding: 0; }
  dl dt {
    font-size: 14px;
    font-weight: bold;
    font-style: italic;
    padding: 0;
    margin: 15px 0 5px; }
    dl dt:first-child {
      padding: 0; }
    dl dt > :first-child {
      margin-top: 0; }
    dl dt > :last-child {
      margin-bottom: 0; }
  dl dd {
    margin: 0 0 15px;
    padding: 0 15px; }
    dl dd > :first-child {
      margin-top: 0; }
    dl dd > :last-child {
      margin-bottom: 0; }

blockquote {
  border-left: 4px solid #dddddd;
  padding: 0 15px;
  color: #777777; }
  blockquote > :first-child {
    margin-top: 0; }
  blockquote > :last-child {
    margin-bottom: 0; }

table {
  padding: 0;border-collapse: collapse; }
  table tr {
    border-top: 1px solid #cccccc;
    background-color: white;
    margin: 0;
    padding: 0; }
    table tr:nth-child(2n) {
      background-color: #f8f8f8; }
    table tr th {
      font-weight: bold;
      border: 1px solid #cccccc;
      margin: 0;
      padding: 6px 13px; }
    table tr td {
      border: 1px solid #cccccc;
      margin: 0;
      padding: 6px 13px; }
    table tr th :first-child, table tr td :first-child {
      margin-top: 0; }
    table tr th :last-child, table tr td :last-child {
      margin-bottom: 0; }

img {
  max-width: 100%; }

span.frame {
  display: block;
  overflow: hidden; }
  span.frame > span {
    border: 1px solid #dddddd;
    display: block;
    float: left;
    overflow: hidden;
    margin: 13px 0 0;
    padding: 7px;
    width: auto; }
  span.frame span img {
    display: block;
    float: left; }
  span.frame span span {
    clear: both;
    color: #333333;
    display: block;
    padding: 5px 0 0; }
span.align-center {
  display: block;
  overflow: hidden;
  clear: both; }
  span.align-center > span {
    display: block;
    overflow: hidden;
    margin: 13px auto 0;
    text-align: center; }
  span.align-center span img {
    margin: 0 auto;
    text-align: center; }
span.align-right {
  display: block;
  overflow: hidden;
  clear: both; }
  span.align-right > span {
    display: block;
    overflow: hidden;
    margin: 13px 0 0;
    text-align: right; }
  span.align-right span img {
    margin: 0;
    text-align: right; }
span.float-left {
  display: block;
  margin-right: 13px;
  overflow: hidden;
  float: left; }
  span.float-left span {
    margin: 13px 0 0; }
span.float-right {
  display: block;
  margin-left: 13px;
  overflow: hidden;
  float: right; }
  span.float-right > span {
    display: block;
    overflow: hidden;
    margin: 13px auto 0;
    text-align: right; }

code, tt {
  margin: 0 2px;
  padding: 0 5px;
  white-space: nowrap;
  border: 1px solid #eaeaea;
  background-color: #f8f8f8;
  border-radius: 3px; }

pre code {
  margin: 0;
  padding: 0;
  white-space: pre;
  border: none;
  background: transparent; }

.highlight pre {
  background-color: #f8f8f8;
  border: 1px solid #cccccc;
  font-size: 13px;
  line-height: 19px;
  overflow: auto;
  padding: 6px 10px;
  border-radius: 3px; }

pre {
  background-color: #f8f8f8;
  border: 1px solid #cccccc;
  font-size: 13px;
  line-height: 19px;
  overflow: auto;
  padding: 6px 10px;
  border-radius: 3px; }
  pre code, pre tt {
    background-color: transparent;
    border: none; }

sup {
    font-size: 0.83em;
    vertical-align: super;
    line-height: 0;
}

kbd {
  display: inline-block;
  padding: 3px 5px;
  font-size: 11px;
  line-height: 10px;
  color: #555;
  vertical-align: middle;
  background-color: #fcfcfc;
  border: solid 1px #ccc;
  border-bottom-color: #bbb;
  border-radius: 3px;
  box-shadow: inset 0 -1px 0 #bbb
}

* {
	-webkit-print-color-adjust: exact;
}
@media screen and (min-width: 914px) {
    body {
        width: 854px;
        margin:0 auto;
    }
}
@media print {
	table, pre {
		page-break-inside: avoid;
	}
	pre {
		word-wrap: break-word;
	}
}
</style>


</head>

<body>

<h1 id="toc_0">Reference-based RNA-seq Analysis in Galaxy and Visualization using UCSC Genome Browser and IGV</h1>

<p>By Hanrui Zhang on [03/02/2021) based on the CSHL Tutorials in Genomics and Bioinformatics: RNA-Seq.    </p>

<p>The workflow in Galaxy was inspired by <a href="https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html#data-upload">this material</a>. </p>

<h2 id="toc_1">RNA-seq Experimental Design</h2>

<ul>
<li>Heterogenous samples (e.g. PDX) will need more reads. Cell lines will typically need less reads.</li>
<li>30-50M PE for analyzing novel transcripts, alternative splicing, heterogeneous samples; 25M PE for DE analysis.</li>
<li>n=4 or more.</li>
</ul>

<h2 id="toc_2">Galaxy Logistics</h2>

<ul>
<li>If registered, history will stay until you delete it.</li>
<li>250GB space. </li>
<li>If need to save space, can delete fastq files once obtaining the BAM file. </li>
<li>People may also upload their data to SRA so that data can be loaded from SRA to galaxy.</li>
</ul>

<h2 id="toc_3">Align to Genome or Transcriptome?</h2>

<h4 id="toc_4">Genome</h4>

<ul>
<li>Advantages: can align novel isoforms</li>
<li>Disadvantages: difficult, spurioous alignments, spliced alignment, gene families, pseudogenes.</li>
<li>Now typical workflow would be to align to the genome</li>
</ul>

<h4 id="toc_5">Transcriptome</h4>

<ul>
<li>Advantages: easy, focused to the part of the genmome that is known to be transcribed</li>
<li>Disadvantages: Reads that come from novel isoforms may not align at all or may be misattributed to a known isoform</li>
</ul>

<h2 id="toc_6">Reference-based RNA-seq Analysis in Galaxy</h2>

<h3 id="toc_7">1. Find, upload, and edit the gtf file (ENSEMBL annotation) for analysis in galaxy</h3>

<h3 id="toc_8">Find the ENSEMBL annotation</h3>

<ul>
<li>Go to the ENSEMBL website</li>
<li>Find the FTP download</li>
<li>Find the annotation needed</li>
<li>Right click and copy the ftp link</li>
<li>Go back to Galaxy and click &quot;upload data&quot; and &quot;Pate/Fetch Data&quot;.</li>
<li>Enter the ftp link, make sure to change the file name that includes the annotation, the file type as &quot;gtf&quot;, amd the annotation e.g. &quot;hg38&quot; (or whatever matchs with the ensemble annotation).</li>
</ul>

<h3 id="toc_9">Use Awk to replace text in specific column:</h3>

<ul>
<li>The chromosome number column in the gtf file need to show &quot;chr1&quot; instead of &quot;1&quot; in order to match the annotation in galaxy. Check the gtf file uploaded and change with awk as stated below if needed.</li>
<li>Search in the tool box for &quot;awk&quot; and find &quot;text formatting with awk&quot;</li>
<li>Select the file to process</li>
<li>In &quot;AWK Program&quot;, put <code>{if ($0 !~ /^#/) { print &quot;chr&quot;$0} else {print $0}}</code></li>
<li>Text formating syntax is difficult, can either find someone to help or try to search in google to see if someone has done the same thing.<br></li>
</ul>

<h3 id="toc_10">2. Upload own data to the galaxy</h3>

<h4 id="toc_11">Uploading own fastq files using ftp upload</h4>

<ul>
<li>Follow this guideline for using client, e.g. FileZilla to connect to the galaxy ftp server <a href="https://galaxyproject.org/ftp-upload/">https://galaxyproject.org/ftp-upload/</a>.</li>
<li>Make sure you download FileZilla from the correct webpage link <a href="https://filezilla-project.org/">https://filezilla-project.org/</a>.</li>
<li>In galaxy, select &quot;upload data&quot; and &quot;choose FTP file&quot;. The files uploaded to the galaxy ftp site via client should appear there.</li>
<li>Can also use cyberduck but it seems not free.</li>
<li>The datasest type should be &quot;fastqsager.gz&quot; in order for HISAT2 in the later steps of the workflow to recognize the file. This can be changed by clicking the pencil icon either here or later step for the Trimmomatic output files. (Delphine&#39;s note: fastqsanger and fastqillumina formats follow the same convention (post ~2010 sequencing will generally))</li>
</ul>

<h3 id="toc_12">3. QC, trimming, QC again, and then alignment to generate BAM file</h3>

<h4 id="toc_13">Grouping fastq file for the same sample</h4>

<ul>
<li>In history, click the checkbox for &quot;Operations on multiple datasets&quot;.</li>
<li>Check the pair of R1 and R2 file for the same sample, click &quot;for the selected datasets&quot; drop menu and select &quot;build dataset pair&quot;, remember to click &quot;swap&quot; to list R1 first. Rename and click &quot;create list&quot;.</li>
</ul>

<h4 id="toc_14">Use fastqc to QC fastq files</h4>

<ul>
<li>Search fastqc tool</li>
<li>“Short read data from your current history”: input datasets selected
with &quot;Dataset Collection&quot;. Select the dataset.</li>
<li>No need to change anything, click &quot;execute&quot;.</li>
</ul>

<h4 id="toc_15">Use Multiqc to generate an aggregated report</h4>

<ul>
<li>Multiqc is to aggregate report together.</li>
<li>“Which tool was used generate logs?”: FastQC</li>
<li>In “FastQC output”: 

<ul>
<li>“Type of FastQC output?”: Raw data</li>
<li>“FastQC output”: Raw data files (output of FastQC tool) – Use
Browse Datasets to find the raw data for FastQC on fastQC read1 et select it (will turn green).</li>
</ul></li>
</ul>

<h4 id="toc_16">Use Trimmomatics to trim adaptor reads</h4>

<ul>
<li>Search tool &quot;Trimmomatic&quot;</li>
<li>“Single-end or paired-end reads?”: Select &quot;Paired-end (as collection)&quot;</li>
<li>Input FASTQ dataset collection with R1/R2 pair: select dataset e.g.
SRR5448889 (will select R1 and R2).</li>
<li>Note: Options to remove adapters

<ul>
<li>Click on Perform initial Illumina-CLIP step? Yes</li>
<li>“Select standard adapter sequences or provide custom?” Standard</li>
<li>“Adapter sequences to use” TruSeq3</li>
</ul></li>
<li>“Output trimmomatic log messages?” yes</li>
<li>Run the tool (Execute). </li>
</ul>

<h4 id="toc_17">Run fastqc and multiqc again</h4>

<h4 id="toc_18">Use HISAT2 to perform alignment and generate BAM file</h4>

<ul>
<li>Search tool &quot;HISAT2&quot;.</li>
<li>&quot;Use a built-in genome&quot;</li>
<li>&quot;Select a reference genome: Mouse: mm10 or Human hg38 as appropriate</li>
<li>Select &quot;Paired-end Dataset Collection&quot;</li>
<li>Choose Paired Collection &quot;Trimmomatic on data * and data *: Paired&quot;.</li>
<li><strong>Specify strand information: make sure to select &quot;Reverse (RF)&quot;</strong></li>
<li><strong>Notes</strong>: can also choose &quot;Paired-end&quot; then select each individual file (the trimmomatic output). The results appear to be the same. </li>
</ul>

<h4 id="toc_19">Visualize BAM file in UCSC genome browser or IGV</h4>

<ul>
<li>Choose the HISAT2 output</li>
<li>Click &quot;display in UCSC main&quot; to visualize the BAM file in UCSC genome browser</li>
<li>Click &quot;display with IGV local&quot; to visualize in IGV, but make sure install and open IGV first. </li>
</ul>

<h4 id="toc_20">Use QualiMap BamQC to perform QC on alignments (bam files) and get summary statistics</h4>

<ul>
<li>Open the QualiMap BamQC tool. Can select multiple HISAT2 output BAM files. Keep all other options as default and execute.</li>
<li>Open the QualiMap Multi-Sample BamQC tool. Select the raw QualiMap BamQC output. For this step, need to select each output file individually and rename - can rename with the sample name.</li>
</ul>

<h3 id="toc_21">4. Generate bigWig file in galaxy for wiggle display in UCSC or IGV</h3>

<ul>
<li>Search tool &quot;bamCoverage&quot;</li>
<li>Select the BAM file used to generate bigWig: &quot;HISAT2 on data * and data *: aligned reads (BAM)</li>
<li>Select the genome: hg38</li>
<li>Bin size: if 1 the bigWig file size will be big; so use default bin size at 50 should usually be good enough.</li>
<li>Everything else can be default as long as the final visualization looks good.</li>
<li>Execute and wait for the job to complete</li>
<li>In galaxy history, find the generated output, click display in &quot;UCSC main&quot; to visualize in UCSC</li>
<li>To overlay multiple tracks, see step 4 http://genome.ucsc.edu/inc/hgCollectionHelpInclude.html </li>
<li><p>Custome track: Can click custom tracks to find the &quot;http&quot; link, that is where the bigWig files are hosted on the galaxy server. The link can be shared with others to visualize the track. </p></li>
<li><p>Save session: Click My data - my session - set up user account and then create a link to share the view of the same session with others.</p></li>
<li><p><strong>Notes:</strong> Files can be sent from galaxy to UCSC, but upload a bigWig file from local drive to UCSC is not possible - the file size is too large and UCSC will not allow it. Therefore, can either use other server, or use IGV to visualize any BAM or bigWig files on the local hard drive.</p></li>
</ul>

<h3 id="toc_22">5. Use htseq-count to count reads overlapping each annotated gene feature</h3>

<ul>
<li>Select the htseq-count tool.</li>
<li>Run the tool in batch mode by selecting “Multiple datasets” (middle icon) for “Aligned SAM/BAM File” (HISAT2 output)</li>
<li>Select the annotation “GFF File” from the history</li>
<li>Other options are by &quot;default&quot; but make sure &quot;stranded&quot; was selected.</li>
<li>Execute</li>
</ul>

<h3 id="toc_23">6. Differential expression analysis using DESeq2</h3>

<p>It is fairly convenient to do DESeq2 analysis in galaxy and obtain the dianostic plotting. But since we have the RNA-seq workflow in R for more flexible and powerful plotting and subsetting, and the newest features in DESeq2, e.g. shrink of the fold change, have not been incorporated in galaxy yet, we should just use our workflow in R for DE analysis.</p>

<h2 id="toc_24">The use of IGV</h2>

<ul>
<li>Open IGV first, make sure the annotation match with the one used in galaxy for analyais, i.e. hg38 (the seletion menu can be find at the left top corner).</li>
<li>Go back to galaxy, find the BAM file or bigWig file, choose display in IGV <strong>local</strong>. Do this one by one for all the BAM or bigWig files.</li>
<li>Rename to &quot;sample name _ coverage&quot; and &quot;sample name _ alignment&quot;</li>
<li>Right click the track, select color alignment by, and select what&#39;s needed, for example  &quot;first of pair strand&quot;.</li>
<li>Can click a read to see the information. Can also right click a read, select &quot;blat read sequence&quot;, a pop-up window comes up, gives information where the sequence align, reassuring the alignment is precise.</li>
<li>The color bar in the middle of the reads means mismatches, we can click the bar to see it. The mismatch could be sequencing error, alignment error, or SNP. If all reads in the position have the same mismatch, which could be PCR duplicate, or sequencing error. </li>
<li>Sashimi plot: right click track and select sashimi plot, select &quot;gene&quot;, select sample to show sashimi plot

<ul>
<li>the number on the top of the line is the number of reads spanning splice junction. The plot helps to identify novel exon (in this case need 30-50 M PE reads) and then a few hundreds reads would be reliable to draw a conclusion.</li>
<li>Can right click to remove a track.</li>
<li>Can save the plot </li>
</ul></li>
<li>Can &quot;save session&quot; to save the format then next time just open session and load the data.</li>
<li>One can obtain anyone&#39;s bigwig file for visualization in IGV. But for UCSC, it has to be a server because the file size is too big for UCSC to allow upload locally. When the workflow used for generating the bigwig files is different the results won&#39;t be quantitatively comparable, though still good for visualization.</li>
</ul>

<h2 id="toc_25">Share visualization of BAM or bigWig files with others</h2>

<h4 id="toc_26">Use &quot;my sessions&quot; in UCSC genome browser to save a session and share the http link with others.</h4>

<ul>
<li><a href="http://genome.ucsc.edu/s/hz2418/mm10_LipaTg4_20200821">http://genome.ucsc.edu/s/hz2418/mm10<em>LipaTg4</em>20200821</a>.<br></li>
</ul>

<h4 id="toc_27">Use CyVerse server to host the BAM or bigWig files and make the track available publically.</h4>

<ul>
<li><a href="http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&amp;hubUrl=https://de.cyverse.org/anon-files/iplant/home/chenyix/trackhub/GENEiPS_scaled/hub.txt">http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&amp;hubUrl=https://de.cyverse.org/anon-files/iplant/home/chenyix/trackhub/GENEiPS_scaled/hub.txt</a></li>
</ul>

<h2 id="toc_28">The use of ENSEMBL and BioMart</h2>

<ul>
<li>ENSEMBL is more gene focused, vs UCSC that is more genome focused with good visualization.</li>
<li>ENSEMBL IDs go beyond 20000000 will most likely be the newly annotated genes, particularly lncRNA and pseudogenes. Well known genes have smaller numbers.</li>
<li>ENSEMBL also has a mobile site. Most people use ENSEMBL works with human and mouse. </li>
<li>ENSEMBL has two other sites: PreEnsembl for other species, some have better annotation than others.</li>
<li>Most people come to ENSEMBL with query need of a certain gene to learn transcripts, protein, homology and orthology search.

<ul>
<li>e.g. can check any genetic variants identified in the gene with scores predicting damaging variants or disease causing variants, and the available data from gnom etc for disease causing variants.</li>
</ul></li>
<li>Always include the ENSEMBL version and ensemble ID in the publication</li>
<li>Example of use of BioMart:

<ul>
<li>Query &quot;Gene Names&quot; to obtain IDs or vice versa</li>
<li>Find all the genes within a certain genomic region</li>
</ul></li>
<li>Steps to use BioMart:

<ul>
<li>Can use the newest version of Ensembl annotation or if needed, can scroll down to the bottom of ensemble home page and click &quot;View in archive site&quot;</li>
<li>Choose the annotation</li>
<li>Click filter and enter the query items and indicate the identify, e.g. stable gene ID or gene name</li>
<li>Click attribution to select the information you want to know.</li>
<li>Click results to see the report and export as needed.</li>
</ul></li>
</ul>

<h2 id="toc_29">Venny</h2>

<ul>
<li>A useful tool to identify overlapped genes from multiple gene list.</li>
<li>Can use ensemble ID, the overlapped list in the venn diagram can be selected and the list can be copy/paste to Biomart to find the gene name.</li>
</ul>

<h2 id="toc_30">Pathway Analysis</h2>

<ul>
<li>The purpose is to guide future experiment and inform hypothesis, therefore the filtering strategy, ranking strategy and the number of genes to include may not matter as much.</li>
<li>For hypergeometric analysis, smaller gene sets and smaller input gene list may lead to only one or two genes in the list are in the pathway still showing statistical significance. This needs to be taken into consideration and and it is generally recommended to prioritize those enriched pathways/gene sets that have more genes from the foreground list.

<ul>
<li><strong>Gene list can be small (one or two genes) therefore looks significant but is not</strong></li>
</ul></li>
<li>It could be informative to visualize the genes in the enriched pathways because it is possible that those genes are all playing roles in a certain part of the large pathway.</li>
<li>Using ENSEMBL ID for query can be helpful to improve </li>
<li>String is a poor-person&#39;s version of IPA

<ul>
<li>Should use ENSEMBL ID to query so that all the top DE genes can be mapped</li>
</ul></li>
<li>DAVID is old and not well updated but easy to use and can

<ul>
<li>Check genes enriched in OMIM: genes associated with disease</li>
</ul></li>
</ul>

<h2 id="toc_31">Notes</h2>

<ul>
<li>A useful resource for track configuration: http://genome.ucsc.edu/inc/hgCollectionHelpInclude.html </li>
<li>Datacamp: Many caourses for learning R/Python; can purchase subscribtion</li>
<li>Codeacademy has good tutorial for commandline tools.</li>
</ul>




</body>

</html>
