<!DOCTYPE html>
<html>

<head>

<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>CSHL_TGB2021</title>


<style type="text/css">
body {
  font-family: Helvetica, arial, sans-serif;
  font-size: 14px;
  line-height: 1.6;
  padding-top: 10px;
  padding-bottom: 10px;
  background-color: white;
  padding: 30px; }

body > *:first-child {
  margin-top: 0 !important; }
body > *:last-child {
  margin-bottom: 0 !important; }

a {
  color: #4183C4; }
a.absent {
  color: #cc0000; }
a.anchor {
  display: block;
  padding-left: 30px;
  margin-left: -30px;
  cursor: pointer;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0; }

h1, h2, h3, h4, h5, h6 {
  margin: 20px 0 10px;
  padding: 0;
  font-weight: bold;
  -webkit-font-smoothing: antialiased;
  cursor: text;
  position: relative; }

h1:hover a.anchor, h2:hover a.anchor, h3:hover a.anchor, h4:hover a.anchor, h5:hover a.anchor, h6:hover a.anchor {
  background: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA09pVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMy1jMDExIDY2LjE0NTY2MSwgMjAxMi8wMi8wNi0xNDo1NjoyNyAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvIiB4bWxuczp4bXBNTT0iaHR0cDovL25zLmFkb2JlLmNvbS94YXAvMS4wL21tLyIgeG1sbnM6c3RSZWY9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9zVHlwZS9SZXNvdXJjZVJlZiMiIHhtcDpDcmVhdG9yVG9vbD0iQWRvYmUgUGhvdG9zaG9wIENTNiAoMTMuMCAyMDEyMDMwNS5tLjQxNSAyMDEyLzAzLzA1OjIxOjAwOjAwKSAgKE1hY2ludG9zaCkiIHhtcE1NOkluc3RhbmNlSUQ9InhtcC5paWQ6OUM2NjlDQjI4ODBGMTFFMTg1ODlEODNERDJBRjUwQTQiIHhtcE1NOkRvY3VtZW50SUQ9InhtcC5kaWQ6OUM2NjlDQjM4ODBGMTFFMTg1ODlEODNERDJBRjUwQTQiPiA8eG1wTU06RGVyaXZlZEZyb20gc3RSZWY6aW5zdGFuY2VJRD0ieG1wLmlpZDo5QzY2OUNCMDg4MEYxMUUxODU4OUQ4M0REMkFGNTBBNCIgc3RSZWY6ZG9jdW1lbnRJRD0ieG1wLmRpZDo5QzY2OUNCMTg4MEYxMUUxODU4OUQ4M0REMkFGNTBBNCIvPiA8L3JkZjpEZXNjcmlwdGlvbj4gPC9yZGY6UkRGPiA8L3g6eG1wbWV0YT4gPD94cGFja2V0IGVuZD0iciI/PsQhXeAAAABfSURBVHjaYvz//z8DJYCRUgMYQAbAMBQIAvEqkBQWXI6sHqwHiwG70TTBxGaiWwjCTGgOUgJiF1J8wMRAIUA34B4Q76HUBelAfJYSA0CuMIEaRP8wGIkGMA54bgQIMACAmkXJi0hKJQAAAABJRU5ErkJggg==) no-repeat 10px center;
  text-decoration: none; }

h1 tt, h1 code {
  font-size: inherit; }

h2 tt, h2 code {
  font-size: inherit; }

h3 tt, h3 code {
  font-size: inherit; }

h4 tt, h4 code {
  font-size: inherit; }

h5 tt, h5 code {
  font-size: inherit; }

h6 tt, h6 code {
  font-size: inherit; }

h1 {
  font-size: 28px;
  color: black; }

h2 {
  font-size: 24px;
  border-bottom: 1px solid #cccccc;
  color: black; }

h3 {
  font-size: 18px; }

h4 {
  font-size: 16px; }

h5 {
  font-size: 14px; }

h6 {
  color: #777777;
  font-size: 14px; }

p, blockquote, ul, ol, dl, li, table, pre {
  margin: 15px 0; }

hr {
  background: transparent url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAYAAAAECAYAAACtBE5DAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAAyJpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvIiB4bWxuczp4bXBNTT0iaHR0cDovL25zLmFkb2JlLmNvbS94YXAvMS4wL21tLyIgeG1sbnM6c3RSZWY9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9zVHlwZS9SZXNvdXJjZVJlZiMiIHhtcDpDcmVhdG9yVG9vbD0iQWRvYmUgUGhvdG9zaG9wIENTNSBNYWNpbnRvc2giIHhtcE1NOkluc3RhbmNlSUQ9InhtcC5paWQ6OENDRjNBN0E2NTZBMTFFMEI3QjRBODM4NzJDMjlGNDgiIHhtcE1NOkRvY3VtZW50SUQ9InhtcC5kaWQ6OENDRjNBN0I2NTZBMTFFMEI3QjRBODM4NzJDMjlGNDgiPiA8eG1wTU06RGVyaXZlZEZyb20gc3RSZWY6aW5zdGFuY2VJRD0ieG1wLmlpZDo4Q0NGM0E3ODY1NkExMUUwQjdCNEE4Mzg3MkMyOUY0OCIgc3RSZWY6ZG9jdW1lbnRJRD0ieG1wLmRpZDo4Q0NGM0E3OTY1NkExMUUwQjdCNEE4Mzg3MkMyOUY0OCIvPiA8L3JkZjpEZXNjcmlwdGlvbj4gPC9yZGY6UkRGPiA8L3g6eG1wbWV0YT4gPD94cGFja2V0IGVuZD0iciI/PqqezsUAAAAfSURBVHjaYmRABcYwBiM2QSA4y4hNEKYDQxAEAAIMAHNGAzhkPOlYAAAAAElFTkSuQmCC) repeat-x 0 0;
  border: 0 none;
  color: #cccccc;
  height: 4px;
  padding: 0;
}

body > h2:first-child {
  margin-top: 0;
  padding-top: 0; }
body > h1:first-child {
  margin-top: 0;
  padding-top: 0; }
  body > h1:first-child + h2 {
    margin-top: 0;
    padding-top: 0; }
body > h3:first-child, body > h4:first-child, body > h5:first-child, body > h6:first-child {
  margin-top: 0;
  padding-top: 0; }

a:first-child h1, a:first-child h2, a:first-child h3, a:first-child h4, a:first-child h5, a:first-child h6 {
  margin-top: 0;
  padding-top: 0; }

h1 p, h2 p, h3 p, h4 p, h5 p, h6 p {
  margin-top: 0; }

li p.first {
  display: inline-block; }
li {
  margin: 0; }
ul, ol {
  padding-left: 30px; }

ul :first-child, ol :first-child {
  margin-top: 0; }

dl {
  padding: 0; }
  dl dt {
    font-size: 14px;
    font-weight: bold;
    font-style: italic;
    padding: 0;
    margin: 15px 0 5px; }
    dl dt:first-child {
      padding: 0; }
    dl dt > :first-child {
      margin-top: 0; }
    dl dt > :last-child {
      margin-bottom: 0; }
  dl dd {
    margin: 0 0 15px;
    padding: 0 15px; }
    dl dd > :first-child {
      margin-top: 0; }
    dl dd > :last-child {
      margin-bottom: 0; }

blockquote {
  border-left: 4px solid #dddddd;
  padding: 0 15px;
  color: #777777; }
  blockquote > :first-child {
    margin-top: 0; }
  blockquote > :last-child {
    margin-bottom: 0; }

table {
  padding: 0;border-collapse: collapse; }
  table tr {
    border-top: 1px solid #cccccc;
    background-color: white;
    margin: 0;
    padding: 0; }
    table tr:nth-child(2n) {
      background-color: #f8f8f8; }
    table tr th {
      font-weight: bold;
      border: 1px solid #cccccc;
      margin: 0;
      padding: 6px 13px; }
    table tr td {
      border: 1px solid #cccccc;
      margin: 0;
      padding: 6px 13px; }
    table tr th :first-child, table tr td :first-child {
      margin-top: 0; }
    table tr th :last-child, table tr td :last-child {
      margin-bottom: 0; }

img {
  max-width: 100%; }

span.frame {
  display: block;
  overflow: hidden; }
  span.frame > span {
    border: 1px solid #dddddd;
    display: block;
    float: left;
    overflow: hidden;
    margin: 13px 0 0;
    padding: 7px;
    width: auto; }
  span.frame span img {
    display: block;
    float: left; }
  span.frame span span {
    clear: both;
    color: #333333;
    display: block;
    padding: 5px 0 0; }
span.align-center {
  display: block;
  overflow: hidden;
  clear: both; }
  span.align-center > span {
    display: block;
    overflow: hidden;
    margin: 13px auto 0;
    text-align: center; }
  span.align-center span img {
    margin: 0 auto;
    text-align: center; }
span.align-right {
  display: block;
  overflow: hidden;
  clear: both; }
  span.align-right > span {
    display: block;
    overflow: hidden;
    margin: 13px 0 0;
    text-align: right; }
  span.align-right span img {
    margin: 0;
    text-align: right; }
span.float-left {
  display: block;
  margin-right: 13px;
  overflow: hidden;
  float: left; }
  span.float-left span {
    margin: 13px 0 0; }
span.float-right {
  display: block;
  margin-left: 13px;
  overflow: hidden;
  float: right; }
  span.float-right > span {
    display: block;
    overflow: hidden;
    margin: 13px auto 0;
    text-align: right; }

code, tt {
  margin: 0 2px;
  padding: 0 5px;
  white-space: nowrap;
  border: 1px solid #eaeaea;
  background-color: #f8f8f8;
  border-radius: 3px; }

pre code {
  margin: 0;
  padding: 0;
  white-space: pre;
  border: none;
  background: transparent; }

.highlight pre {
  background-color: #f8f8f8;
  border: 1px solid #cccccc;
  font-size: 13px;
  line-height: 19px;
  overflow: auto;
  padding: 6px 10px;
  border-radius: 3px; }

pre {
  background-color: #f8f8f8;
  border: 1px solid #cccccc;
  font-size: 13px;
  line-height: 19px;
  overflow: auto;
  padding: 6px 10px;
  border-radius: 3px; }
  pre code, pre tt {
    background-color: transparent;
    border: none; }

sup {
    font-size: 0.83em;
    vertical-align: super;
    line-height: 0;
}

kbd {
  display: inline-block;
  padding: 3px 5px;
  font-size: 11px;
  line-height: 10px;
  color: #555;
  vertical-align: middle;
  background-color: #fcfcfc;
  border: solid 1px #ccc;
  border-bottom-color: #bbb;
  border-radius: 3px;
  box-shadow: inset 0 -1px 0 #bbb
}

* {
	-webkit-print-color-adjust: exact;
}
@media screen and (min-width: 914px) {
    body {
        width: 854px;
        margin:0 auto;
    }
}
@media print {
	table, pre {
		page-break-inside: avoid;
	}
	pre {
		word-wrap: break-word;
	}
}
</style>


</head>

<body>

<h1 id="toc_0">Reference-based RNA-seq Analysis in Galaxy, and Data Visualization in IGV and UCSC Genome Browser</h1>

<p>By Hanrui Zhang on [12/01/2021] based on the CSHL Tutorials in Genomics and Bioinformatics: RNA-Seq and <a href="https://bioinformaticsdotca.github.io/EPI_2021_Module5_lab">bioinformatics.ca</a> workflow.    </p>

<p>The workflow in Galaxy was inspired by <a href="https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html#data-upload">this material</a>. </p>

<h2 id="toc_1">1. Tools Needed for this Workflow</h2>

<ul>
<li>Register an account in Galaxy, an open-source, web-based platform for data-intensive biomedical research: <a href="https://usegalaxy.org/">https://usegalaxy.org/</a>.</li>
<li>Download and install the FileZilla Client, the free solution for File Transfer Protocol (FTP): <a href="https://filezilla-project.org/">https://filezilla-project.org/</a>.</li>
<li>Download and install the Integrative Genomics Viewer (IGV): <a href="https://software.broadinstitute.org/software/igv/download">https://software.broadinstitute.org/software/igv/download</a>.</li>
</ul>

<h2 id="toc_2">2. Background</h2>

<h3 id="toc_3">2.1 RNA-seq Experimental Design</h3>

<ul>
<li>Heterogenous samples (e.g. patient-derived xenograft tumor model) will need more reads. Cell lines will typically need fewer reads.</li>
<li>30-50M paired-end (PE) reads for analyzing novel transcripts, alternative splicing, heterogeneous samples; 25M PE for differential expression (DE) analysis.</li>
<li>n=4 or more.</li>
</ul>

<h3 id="toc_4">2.2 Galaxy Logistics</h3>

<ul>
<li>Register and access: <a href="https://usegalaxy.org/">https://usegalaxy.org/</a></li>
<li>If registered, history will stay until you delete it.</li>
<li>250GB space. </li>
<li>If need to save space, can delete fastq files once obtaining the BAM file. </li>
<li>People may also upload their data to Sequence Read Archive (<a href="https://www.ncbi.nlm.nih.gov/sra">SRA</a>, the largest publicly available repository of high throughput sequencing data) so that data can be loaded from SRA to the galaxy.</li>
</ul>

<h3 id="toc_5">2.3 Align to the Genome or the Transcriptome?</h3>

<ul>
<li><p>Genome</p>

<ul>
<li>Advantages: can align novel isoforms.</li>
<li>Disadvantages: difficult, spurious alignments, spliced alignment, gene families, pseudogenes.</li>
<li>Now typical workflow would be to align to the genome.</li>
</ul></li>
<li><p>Transcriptome</p>

<ul>
<li>Advantages: easy, focused on the part of the genome that is known to be transcribed</li>
<li>Disadvantages: Reads that come from novel isoforms may not align at all or may be misattributed to a known isoform.</li>
</ul></li>
</ul>

<h2 id="toc_6">3. Reference-based RNA-seq Analysis in Galaxy</h2>

<h3 id="toc_7">3.1 ANALYZING A TOY DATASET (<a href="https://bioinformaticsdotca.github.io/EPI_2021_Module5_lab">REFERENCE</a>):</h3>

<h4 id="toc_8">3.1.1. Create a new history, maybe name it as &quot;Exercise&quot;.</h4>

<h4 id="toc_9">3.1.2. Upload the dataset:</h4>

<ul>
<li>For this exercise, we’ll use subsets of data from the Illumina BodyMap 2.0 project, from human adrenal gland tissues. The sampled reads are paired-end 50bp that map mostly to a 500Kb region of chromosome 19, positions 3-3.5 million (chr19:3000000:3500000). (source: https://usegalaxy.org/u/jeremy/p/galaxy-rna-seq-analysis-exercise)</li>
<li>Import the following two FASTQ files in your user space. To do so:

<ul>
<li>Click the Upload Data button on the upper left of the screen</li>
<li>In the popup, click on the <code>Paste/Fetch data</code> button</li>
<li>You can provide both URLs in the same text box
<code>https://raw.githubusercontent.com/bioinformatics-ca/EPI_2021/master/files/adrenal_1.fastq</code>
<code>https://raw.githubusercontent.com/bioinformatics-ca/EPI_2021/master/files/adrenal_2.fastq</code></li>
<li>Under Genome (set all), specify <code>Human Feb. 2009 (GRCh37/hg19) (hg19)</code>.</li>
<li>Click on <code>Start</code>.</li>
</ul></li>
<li>After it finished uploading (green state), you can rename the two imported files (if desired), for better organization. </li>
</ul>

<h4 id="toc_10">3.1.3. Running the FastQC tool for quality control (QC) on the fastq files (reads off the sequencer):</h4>

<ul>
<li>To find it, use the search window at the top of the Tools column (left panel).</li>
<li>From the <code>FastQC</code> tool interface, for the field <code>Short read data from your current history</code>, choose adrenal_1.</li>
<li>Click on <code>Execute</code>.</li>
<li>Pay attention to the green notice, which provides details about the input and output of the job you just launched.</li>
<li>Once the job is completed, examine the Webpage results from the history bar using the <code>Eye</code> icon.</li>
<li>Raw output statistics are also available, and can also be seen with the <code>Eye</code> icon.</li>
<li>Repeat the same operations for adrenal_2. As a shortcut, you can click on the FastQC history item, then click on the <code>Run this job again</code> icon and simply change the input file to automatically reuse the same parameters.</li>
</ul>

<h4 id="toc_11">3.1.4. Trim the reads with Trimmomatic tool to improve the quality of the dataset by removing bad quality bases, clipping adapters and so on:</h4>

<ul>
<li>Set the input as Paired-end (two separate input files)</li>
<li>Give the adrenal<em>1 file for direction 1, and adrenal</em>2 for direction 2.</li>
</ul>

<h4 id="toc_12">3.1.5. Run FastQC again on both <strong>paired</strong> files, and compare results with pre-trimming FastQC output.</h4>

<ul>
<li>Can enable &quot;scratchbook&quot; to visualize FastQC reports pre- and post-trimming.</li>
</ul>

<h4 id="toc_13">3.1.6. Use HISAT2 to align the reads to the human genome, output is a BAM file</h4>

<ul>
<li>Search and select the HISAT2 tool</li>
<li>For the <code>Select a reference genome</code>, search <code>hg19</code> and select <code>Human (Homo sapiens) (b37): hg19</code>.</li>
<li>For <code>Is this a single or paired library</code>, select `paired-end.</li>
<li>For <code>FASTA/G file #1</code>, select &quot;(R1 paired); for <code>FASTA/Q file #2</code>, select &quot;(R2 paired).</li>
<li>Click <code>execute</code>.</li>
<li>The output is the BAM file (*.bam), which is the compressed binary version of a SAM file that is used to represent aligned sequences up to 128 Mb.</li>
<li>Visualize the BAM file in UCSC genome browser:

<ul>
<li>Click the HISAT2 output file name, then click <code>display at UCSC main</code>. This will open another browser window for the UCSC genome browser.</li>
<li>What do you see now? Do you see the BAM file track?</li>
<li>Enter <code>chr19:3,478,706-3,508,735</code>.</li>
<li>Under &quot;Custom Tracks&quot;, choose between &quot;full&quot; and other options to visualize the alignment.</li>
</ul></li>
<li>If you open IGV, you may also click <code>display with IGV local Human hg19</code> to visualize the data in IGV locally.</li>
</ul>

<h4 id="toc_14">3.1.7. Use the bamCoverage tool to create a bigWig coverage track:</h4>

<ul>
<li>Search the <code>bamCoverage</code> tool</li>
<li>Select the &quot;BAM&quot; file.</li>
<li>The output will be bigwig file</li>
<li>Choose the bigwig output file, click <code>display at UCSC main</code>. Now visualize the bigwig coverage file.</li>
</ul>

<h4 id="toc_15">Next we will move to the real datasets. We will do the above steps, and we will also add using HTSeq to count reads for each gene.</h4>

<h3 id="toc_16">3.2 ANALYZING A REAL DATASET:</h3>

<h3 id="toc_17">3.1.1. Create a new history.</h3>

<p>We will analyze a real dataset from Ldlr-/- mice sequenced at 20M 100bp PE. We will use annotation file (Ensembl 102: Nov 2020 (GRCm38.p6)).</p>

<h3 id="toc_18">3.2.2. Find, upload, and edit the gtf file (ENSEMBL annotation) for analysis in the Galaxy</h3>

<h4 id="toc_19">Find the ENSEMBL annotation</h4>

<ul>
<li>Explanation about <a href="https://m.ensembl.org/info/website/upload/gff.html">gtf file</a></li>
<li>Go to the ENSEMBL site <a href="https://useast.ensembl.org/index.html">https://useast.ensembl.org/index.html</a>.</li>
<li>Find <code>Download</code>, select <code>FTP download</code>, find the current gtf or previous release in the archived sites. </li>
<li>Find the annotation needed. e.g.<br>
for human <code>homo_sapiens/Homo_sapiens.GRCh38.104.chr.gtf.gz</code>;<br>
for mice <code>mus_musculus/Mus_musculus.GRCm39.104.chr.gtf.gz</code>.</li>
<li>Right-click and copy the FTP link. <strong>For this exercise, use this ftp link <code>http://ftp.ensembl.org/pub/release-102/gtf/mus_musculus/Mus_musculus.GRCm38.102.chr.gtf.gz</code></strong>. </li>
<li>Go back to Galaxy and click &quot;upload data&quot; and &quot;Paste/Fetch Data&quot;.</li>
<li>Enter the FTP link, make sure to change the file name that includes the annotation (GRCm38.102), the file type as &quot;gtf&quot;, and the annotation e.g. &quot;GRCm38B&quot; (or other species matching with the ensemble annotation).</li>
<li>Click &quot;Start&quot; to upload the annotation.</li>
</ul>

<h4 id="toc_20">Use Awk to replace text in a specific column:</h4>

<ul>
<li>The chromosome number column in the gtf file needs to show &quot;chr1&quot; instead of &quot;1&quot; in order to match the annotation in the galaxy. Check the gtf file uploaded and change with awk as stated below if needed.</li>
<li>Search in the toolbox for &quot;awk&quot; and find &quot;text formatting with awk&quot;</li>
<li>Select the file to process</li>
<li>In &quot;AWK Program&quot;, put <code>{if ($0 !~ /^#/) { print &quot;chr&quot;$0} else {print $0}}</code></li>
<li>Text formatting syntax is difficult, can either find someone to help or try to search in google to see if someone has done the same thing.<br></li>
</ul>

<h3 id="toc_21">3.2.3. Upload own data to the galaxy</h3>

<h4 id="toc_22">Uploading own fastq files using ftp upload</h4>

<ul>
<li>Make sure you download FileZilla from the correct webpage link <a href="https://filezilla-project.org/">https://filezilla-project.org/</a>.</li>
<li>Follow this guideline for using client, e.g. FileZilla to connect to the galaxy FTP server <a href="https://galaxyproject.org/ftp-upload/">https://galaxyproject.org/ftp-upload/</a>.

<ul>
<li>File - my site</li>
<li>New site</li>
<li>provide a name, e.g. &quot;galaxy&quot;</li>
<li>Protocol: File Transfer Protocol</li>
<li>Host: usegalaxy.org</li>
<li>Encryption: Require explicit FTP over TLS</li>
<li>Logon Type: Ask for a password</li>
<li>User: same as for log in the galaxy</li>
<li>Click &quot;Connect&quot;</li>
<li>&quot;Accept&quot; the agreement</li>
<li>Once connected, can drag files from the local folder to the server (left to right). The bottom panel in FileZilla will show the progress of uploading.</li>
<li>Once the uploading is completed, go back to the galaxy.</li>
</ul></li>
<li>In the galaxy, select &quot;upload data&quot; and &quot;choose remote file&quot; - &quot;FTP directory&quot;. The files uploaded to the galaxy FTP site via client should appear there.</li>
<li>Can also use Cyberduck (available through CUIT).</li>
<li>The dataset type should be &quot;fastqsager.gz&quot; in order for HISAT2 in the later steps of the workflow to recognize the file. This can be changed by clicking the pencil icon either here or later step for the Trimmomatic output files. (Delphine&#39;s note: fastqsanger and fastqillumina formats follow the same convention (post ~2010 sequencing will generally))</li>
</ul>

<h3 id="toc_23">3.2.4. QC, trimming, QC again, and then alignment to generate BAM file</h3>

<h4 id="toc_24">Grouping fastq file for the same sample</h4>

<ul>
<li>In history, click the checkbox for &quot;Operations on multiple datasets&quot;.</li>
<li>Check the pair of R1 and R2 file for the same sample, click &quot;for the selected datasets&quot; drop menu and select &quot;build dataset pair&quot;, remember to click &quot;swap&quot; to list R1 first. Rename (e.g. to the sample name) and click &quot;create list&quot;.</li>
</ul>

<h4 id="toc_25">Use fastqc to QC fastq files</h4>

<ul>
<li>Search &quot;FastQC&quot; tool</li>
<li>Find “Short read data from your current history”: Click &quot;Dataset Collection&quot; as the input dataset. Select the dataset pair prepared in the previous step.</li>
<li>No need to change anything, click &quot;execute&quot;.</li>
</ul>

<h4 id="toc_26">Use Multiqc to generate an aggregated report</h4>

<ul>
<li>Multiqc is to aggregate reports together.</li>
<li>Search the &quot;MultiQC&quot; tool and select</li>
<li>For “Which tool was used to generate logs?”, select &quot;FastQC&quot;</li>
<li>In “FastQC output”: 

<ul>
<li>For “Type of FastQC output?”, select &quot;Raw data&quot;</li>
<li>For “FastQC output”, select &quot;Raw data files (output of FastQC tool) – Use
<code>Browse Datasets</code> to find the raw data for FastQC on fastQC read1 et select it (will turn green). Press OK.</li>
<li>Execute</li>
<li>Examine the MultiQC output.</li>
</ul></li>
</ul>

<h4 id="toc_27">Use Trimmomatics to trim adaptor reads</h4>

<ul>
<li>Search tool &quot;Trimmomatic&quot;</li>
<li>“Single-end or paired-end reads?”: Select &quot;Paired-end (as collection)&quot;</li>
<li>Input FASTQ dataset collection with R1/R2 pair: select dataset e.g.
SRR5448889 (will select R1 and R2).</li>
<li>Note: Options to remove adapters

<ul>
<li>Click on Perform initial Illumina-CLIP step? Yes</li>
<li>“Select standard adapter sequences or provide custom?” Standard</li>
<li>“Adapter sequences to use” TruSeq3</li>
</ul></li>
<li>“Output trimmomatic log messages?” yes</li>
<li>Run the tool (Execute). </li>
</ul>

<h4 id="toc_28">Run fastqc and multiqc again</h4>

<h4 id="toc_29">Use HISAT2 to perform alignment and generate BAM file</h4>

<ul>
<li>Search tool &quot;HISAT2&quot;.</li>
<li>&quot;Use a built-in genome&quot;</li>
<li>&quot;Select a reference genome: Mouse: mm10 or Human hg38 as appropriate</li>
<li>Select &quot;Paired-end Dataset Collection&quot;</li>
<li>Choose Paired Collection &quot;Trimmomatic on data * and data *: Paired&quot;.</li>
<li><strong>Specify strand information: make sure to select &quot;Reverse (RF)&quot;</strong></li>
<li><strong>Notes</strong>: can also choose &quot;Paired-end&quot; then select each individual file (the trimmomatic output). The results appear to be the same. </li>
</ul>

<h4 id="toc_30">Visualize BAM file in UCSC genome browser or IGV</h4>

<ul>
<li>Choose the HISAT2 output</li>
<li>Click &quot;display in UCSC main&quot; to visualize the BAM file in the UCSC genome browser</li>
<li>Click &quot;display with IGV local&quot; to visualize in IGV, but make sure to install and open IGV first. </li>
</ul>

<h4 id="toc_31">Use QualiMap BamQC to perform QC on alignments (bam files) and get summary statistics</h4>

<ul>
<li>Open the &quot;QualiMap BamQC&quot; tool. Can select multiple HISAT2 output BAM files. Keep all other options as default and execute.</li>
<li>Open the QualiMap Multi-Sample BamQC tool. Select the raw QualiMap BamQC output. For this step, need to select each output file individually and rename - can rename with the sample name.</li>
</ul>

<h3 id="toc_32">3.2.5. Generate bigWig file in the galaxy for wiggle display in UCSC or IGV</h3>

<ul>
<li>Search tool &quot;bamCoverage&quot;</li>
<li>Select the BAM file used to generate bigWig: &quot;HISAT2 on data * and data *: aligned reads (BAM)</li>
<li>Select the genome: e.g. mm10</li>
<li>Bin size: if 1 the bigWig file size will be big; so using default bin size at 50 should usually be good enough.</li>
<li>Everything else can be default as long as the final visualization looks good.</li>
<li>Execute and wait for the job to complete</li>
<li>In galaxy history, find the generated output, click display in &quot;UCSC main&quot; to visualize in UCSC</li>
<li>To overlay multiple tracks, see step 4 http://genome.ucsc.edu/inc/hgCollectionHelpInclude.html </li>
<li><p>Custome track: Can click custom tracks to find the &quot;http&quot; link, which is where the bigWig files are hosted on the galaxy server. The link can be shared with others to visualize the track. </p></li>
<li><p>Save session: Click My data - my session - set up a user account and then create a link to share the view of the same session with others.</p></li>
<li><p><strong>Notes:</strong> Files can be sent from galaxy to UCSC, but uploading a bigWig file from a local drive to UCSC is not possible - the file size is too large and UCSC will not allow it. Therefore, can either use other servers or use IGV to visualize any BAM or bigWig files on the local hard drive.</p></li>
</ul>

<h3 id="toc_33">3.2.6. Use htseq-count to count reads overlapping each annotated gene feature</h3>

<ul>
<li>Select the htseq-count tool.</li>
<li>Run the tool in batch mode by selecting “Multiple datasets” (middle icon) for “Aligned SAM/BAM File” (HISAT2 output)</li>
<li>Select the annotation “GFF File” from the history</li>
<li>Other options are by &quot;default&quot; but make sure &quot;stranded&quot; was selected.</li>
<li>Execute</li>
</ul>

<h3 id="toc_34">3.2.7. Differential expression analysis using DESeq2</h3>

<p>It is fairly convenient to do DESeq2 analysis in the galaxy and obtain the diagnostic plotting. But since we have the RNA-seq workflow in R for more flexible and powerful plotting and subsetting, and the newest features in DESeq2, e.g. shrink of the fold change, have not been incorporated in galaxy yet, we should just use our workflow in R for DE analysis.</p>

<h3 id="toc_35">3.2.8. Sharing your History</h3>

<ul>
<li>Open the History Options menu (gear icon) at the top of your history panel. Make History accessible. A Share Link will appear that you give to others.</li>
<li>Anybody who has this link can view and copy your history.</li>
</ul>

<h2 id="toc_36">4. Data Visualization in IGV and <a href="https://genome.ucsc.edu/">UCSC Genome Browser</a></h2>

<h3 id="toc_37">Data Visualization in IGV</h3>

<ul>
<li>Open IGV first, make sure the annotation match with the one used in the Galaxy for analysis, i.e. hg38 (the selection menu can be found at the left top corner).</li>
<li>Go back to the galaxy, find the BAM file or bigWig file, choose display in IGV <strong>local</strong>. Do this one by one for all the BAM or bigWig files.</li>
<li>Rename to &quot;sample name _ coverage&quot; and &quot;sample name _ alignment&quot;</li>
<li>Right-click the track, select color alignment by, and select what&#39;s needed, for example, &quot;first of pair strand&quot;.</li>
<li>Can click a read to see the information. Can also right-click a read, select &quot;blat read sequence&quot;, a pop-up window comes up, giving information where the sequence aligns, reassuring the alignment is precise.</li>
<li>The color bar in the middle of the reads means mismatches, we can click the bar to see it. The mismatch could be sequencing error, alignment error, or SNP. If all reads in the position have the same mismatch, which could be PCR duplicate or sequencing error. </li>
<li>Sashimi plot: right-click track and select sashimi plot, select &quot;gene&quot;, select sample to show sashimi plot

<ul>
<li>the number on the top of the line is the number of reads spanning splice junction. The plot helps to identify novel exon (in this case needs 30-50 M PE reads) and then a few hundred reads would be reliable to draw a conclusion.</li>
<li>Can right-click to remove a track.</li>
<li>Can save the plot </li>
</ul></li>
<li>Can &quot;save session&quot; to save the format then next time just open session and load the data.</li>
<li>One can obtain anyone&#39;s bigwig file for visualization in IGV. But for UCSC, it has to be a server because the file size is too big for UCSC to allow upload locally. When the workflow used for generating the bigwig files is different the results won&#39;t be quantitatively comparable, though still good for visualization.</li>
</ul>

<h3 id="toc_38">Share visualization of BAM or bigWig files with others in UCSC Genome Browser</h3>

<ul>
<li><p>Use &quot;my sessions&quot; in the UCSC genome browser to save a session and share the URL with others: e.g.<a href="http://genome.ucsc.edu/s/hz2418/mm10_LipaTg4_20200821">http://genome.ucsc.edu/s/hz2418/mm10<em>LipaTg4</em>20200821</a>.     </p></li>
<li><p>Use CyVerse server to host the BAM or bigWig files and make the track available publically: <a href="http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&amp;hubUrl=https://de.cyverse.org/anon-files/iplant/home/chenyix/trackhub/GENEiPS_scaled/hub.txt">http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&amp;hubUrl=https://de.cyverse.org/anon-files/iplant/home/chenyix/trackhub/GENEiPS_scaled/hub.txt</a></p></li>
</ul>

<h2 id="toc_39">5. Other Useful Tools and Resources for Data Interpretation after Completing the Initial Analyses</h2>

<h3 id="toc_40">Using ENSEMBL and BioMart to pull out information for the gene-of-interest</h3>

<ul>
<li><p><a href="https://useast.ensembl.org/index.html">ENSEMBL</a>: </p>

<ul>
<li>ENSEMBL is more gene-focused. UCSC is more genome-focused with good visualization.</li>
<li>ENSEMBL IDs that go beyond 20000000 will most likely be the newly annotated genes, particularly lncRNA and pseudogenes. Well-known genes have smaller numbers. <strong>Always include the ENSEMBL version and ensemble ID in the publication.</strong></li>
<li>Most people use ENSEMBL while working with human and mouse data. For other species, some have better annotation than others.</li>
<li>Most people come to ENSEMBL with query needs of a specific gene(s) to learn about transcripts, protein, homology, and orthology search.<br>
e.g. can check any genetic variants identified in the gene with scores predicting damaging variants or disease-causing variants, and the available data from gnom, etc. for disease-causing variants.</li>
<li>ENSEMBL has an Archive Site so that there are stable links to data from a particular release.</li>
<li>There is also a mobile site <a href="https://m.ensembl.org/index.html">https://m.ensembl.org/index.html</a>.</li>
</ul></li>
<li><p><a href="http://useast.ensembl.org/biomart/martview/41523e21a42e9f6f166fba5a92de35e3">BioMart</a>:</p>

<ul>
<li>Example of using BioMart -

<ul>
<li>Query &quot;Gene Names&quot; to obtain IDs or vice versa</li>
<li>Find all the genes within a certain genomic region</li>
</ul></li>
<li>Steps to use BioMart:

<ul>
<li>Can use the newest version of Ensembl annotation or if needed, can scroll down to the bottom of ensemble home page and click &quot;View in archive site&quot;</li>
<li>Choose the annotation</li>
<li>Click &quot;Filters&quot; and enter the query items and indicate the identity, e.g. stable gene ID or gene name</li>
<li>Click &quot;Attributes&quot; to select the information you want to know.</li>
<li>Click &quot;Results&quot; to see the report and export as needed.</li>
</ul></li>
</ul></li>
</ul>

<h3 id="toc_41">Using <a href="https://bioinfogp.cnb.csic.es/tools/venny/">Venny</a> to draw Venn Diagrams from lists of genes</h3>

<ul>
<li>A useful tool to identify overlapped genes from multiple gene lists.</li>
<li>Can use ensemble ID, the overlapped list in the Venn diagram can be selected and the list can be copied/pasted to Biomart to find the gene name.</li>
</ul>

<h3 id="toc_42">Pathway Analysis</h3>

<ul>
<li>The purpose is to guide future experiments and inform hypotheses, therefore the filtering strategy, ranking strategy and the number of genes to include may not matter as much.</li>
<li><p>For hypergeometric analysis, smaller gene sets and smaller input gene lists may lead to only one or two genes in the list being in the pathway still showing statistical significance. This needs to be taken into consideration and it is generally recommended to prioritize those enriched pathways/gene sets that have more genes from the foreground list.</p>

<ul>
<li>Gene list can be small (one or two genes) therefore looks significant but is not.</li>
</ul></li>
<li><p>Using ENSEMBL ID for queries can be helpful to improve mapping.</p></li>
<li><p><a href="https://string-db.org/">String</a> is the poor person&#39;s version of IPA.</p>

<ul>
<li>Should use ENSEMBL ID to query so that all the top DE genes can be mapped</li>
</ul></li>
<li><p>It could be informative to visualize the genes in the enriched pathways because it is possible that those genes are all playing roles in a certain part of the large pathway.          </p></li>
<li><p><a href="https://david.ncifcrf.gov/">DAVID</a> is old but is getting better at being updated. It is easy to use and can visualize genes on pathway maps.</p></li>
<li><p>Check genes enriched in OMIM: genes associated with diseases.</p></li>
</ul>

<h2 id="toc_43">6. Additional Notes</h2>

<ul>
<li>A useful resource for track configuration: http://genome.ucsc.edu/inc/hgCollectionHelpInclude.html </li>
<li>Datacamp: Many courses for learning R/Python; can purchase a subscription.</li>
<li>Codeacademy has good tutorials for command-line tools.</li>
</ul>




</body>

</html>
